expression/builtin_other.go

```diff
@@ -106,11 +117,51 @@ func (c *inFunctionClass) getFunction(ctx sessionctx.Context, args []Expression)
 // builtinInIntSig see https://dev.mysql.com/doc/refman/5.7/en/comparison-operators.html#function_in
 type builtinInIntSig struct {
 	baseBuiltinFunc
+	args      []Expression
+	hashSet   map[int64]bool
+	threshold int
+}
+
+func (b *builtinInIntSig) buildHashMapForConstArgs(ctx sessionctx.Context) error {
+	fmt.Println("enter buildHashMapForConstArgs")
+	b.args = make([]Expression, 0, len(b.baseBuiltinFunc.args))
```

Comment:
You could not change the args slice. The in function may be sent to TiKV or used to build range to apply a index scan. In such case, we need to read its args.
You can add a field like `nonConstItems`.